gcloud/datastore/batch.py

```diff
@@ -209,15 +210,7 @@ def put(self, entity):
         if entity.key is None:
             raise ValueError("Entity must have a key")
 
-        key_pb = entity.key.to_protobuf()
-        properties = dict(entity)
-        exclude = tuple(entity.exclude_from_indexes)
-
-        self.connection.save_entity(
-            self.dataset_id, key_pb, properties,
-            exclude_from_indexes=exclude, mutation=self.mutation)
-
-        if entity.key.is_partial:
+        if _assign_entity_to_mutation(self.mutation, entity):
```

Comment:
Small nit: the name `_assign_entity_to_mutation` doesn't seem to indicate the return value will mean auto-id assigned. Would it be more intuitive to pass `self._auto_id_entities` as an argument?